### PR TITLE
AWS Cognito as idP for Kubeflow v1.3

### PR DESCRIPTION
**Pull request by surajkota**
### Description of your changes:
- Add support for deploying Kubeflow with AWS Cognito as an identity provider.
  - Add an EnvoyFilter to inject the `kubeflow-userid` header for the central dashboard to match the profile of an authenticated user.
  - Fix for alb-ingress-controller deployment to fetch `clusterName` from `aws-alb-ingress-controller-config` ConfigMap value
  - Fix to create annotations for ingress with values fetched from `istio-ingress-cognito-parameters` ConfigMap '

### Pending
README with Instructions on deploying KF with Cognito. Will create a separate PR

### Testing
- Testing done using kustomize installation

### CODE REVIEW

File: distributions/kfdef/kfctl_aws.v1.3.0.yaml [hunk 1]
```diff
@@ -22,18 +22,13 @@ spec:
   - kustomizeConfig:
       repoRef:
         name: manifests
-        path: common/cert-manager/cert-manager-kube-system-resources/base
```
Comment (rrrkharse) [inline distributions/kfdef/kfctl_aws.v1.3.0.yaml:25]: since we are not supporting using kfctl for 1.3 I think we should leave this file unchanged

**Comment by rrrkharse** [comment]
otherwise lgtm

File: distributions/kfdef/kfctl_aws.v1.3.0.yaml (hunk 1 above)
Comment (surajkota) [reply distributions/kfdef/kfctl_aws.v1.3.0.yaml:25]: This file was already published by PatricXYZ when 1.3.0 was released. So I am just keeping it up to date with the branch i.e. 1.3.1.

I suggest either we remove it completely or keep this working version

Comment (goswamig) [reply distributions/kfdef/kfctl_aws.v1.3.0.yaml:25]: Lets remove it.

Comment (goswamig) [reply distributions/kfdef/kfctl_aws.v1.3.0.yaml:25]: Lets remove this and all kfctl references

**Review by goswamig — APPROVED** [review]

File: distributions/kfdef/kfctl_aws.v1.3.0.yaml (hunk 1 above)
Comment (surajkota) [reply distributions/kfdef/kfctl_aws.v1.3.0.yaml:30]: @rrrkharse @goswamig done, please take another look

**Review by goswamig — APPROVED** [review]

**Pull request merged**